cmd/aws-vpc-cni/main.go

```diff
@@ -217,14 +217,14 @@ func isValidJSON(inFile string) error {
 	return json.Unmarshal([]byte(inFile), &result)
 }
 
-func generateJSON(jsonFile string, outFile string) error {
+func generateJSON(jsonFile string, outFile string, getNodePrimaryAddress func() (string, error)) error {
 	byteValue, err := ioutil.ReadFile(jsonFile)
 	if err != nil {
 		return err
 	}
 
 	var nodeIP string
-	nodeIP, err = getNodePrimaryV4Address()
+	nodeIP, err = getNodePrimaryAddress()
```

Comment:
Do we send anything other than `getNodePrimaryV4Address` function to this ?

Reply:
No, I made the function a parameter only to simplify mock testing, and I dropped the V4 only to have a different name, and since Go does not support default function parameters.

Reply:
Adding this as a parameter to `generateJSON` function is breaking the modularity too.
I assume this was done writing the test case.

The the best approach will be to introduce another function and keep the generateJSON to taking an "infile" and "outfile" - it just much to reason about while reading the code.

Reply:
Yeah this was done for mocking. Not sure I follow the other function approach, can you expand? As long as `generateJSON` makes an imds call, it needs mocking

Reply:
I missed seeing  `generateJSON` is calling `getNodePrimaryV4Address` even now.

- In that case, moving it parameter is fine for testing.

Should we break that call out of the generateJSON itself, and do the call before calling generateJSON and send node-ip if the call was successful?

```
	var nodeIP string
	nodeIP, err = getNodePrimaryV4Address()
	if err != nil {
		log.Errorf("Failed to get Node IP")
		return err
	}

    generateJSON(nodeIp string, sonFile string, outFile string)
```


Either this, or the the current approach of sending a function parameter is fine.
But former looks a bit more readable to me.

Reply:
Yeah, that makes these more testable separately, so I will make that change

Reply:
updated in latest commit